sw/airborne/peripherals/mpu9250.c

```diff
@@ -110,6 +114,27 @@ void mpu9250_send_config(Mpu9250ConfigSet mpu_set, void *mpu, struct Mpu9250Conf
       mpu_set(mpu, MPU9250_REG_INT_ENABLE, (config->drdy_int_enable << 0));
       config->init_status++;
       break;
+
```

Comment:
The i2c master clock, delay and enable is already configured in mpu9250_configure_i2c_slaves of mpu9250_spi.c, so I don't see why you would do it again afterwards...
Also you are trying to set clock and p_nsr in user_ctrl reg instead of I2C_MST_CTRL

If the P_NSR is needed (stop instead of restart between reads) it should be added as an option to the config struct and set in mpu9250_configure_i2c_slaves together with the clock.